query-result-writer.hh

```diff
@@ -136,9 +136,9 @@ public:
             return stop_iteration::no;
```

Comment:
An alternative is to have the coordinator send a very large (or even infinite) tombstone limit.

I don't see a real difference between the two. At the coordinator it's a little bit more aesthetically pleasing maybe since we don't send a small value and then ignore it.

Reply:
Trying to address this at the coordinator side has a higher chance to miss some cases. Imagine this as a funnel: the coordinator is the wide part of the funnel, the replica side touched by this patch is the narrow-end, which is a single method. Patching this single method automatically enforces the new behaviour for all users.

Furthermore, I think un-paged queries being exempt from the tombstone-limit is a simple enough exception.

Reply:
Makes sense.